datalad/support/annexrepo.py

```diff
@@ -245,7 +249,19 @@ def __init__(self, path, runner=None,
             # '' cannot be converted to int (via Constraint as defined for
             # "datalad.repo.version" in common_cfg
             # => Allow conversion to result in None?
-            if not version:
+            if version:
+                try:
+                    version = int(version)
+                except ValueError:
+                    # Just give a warning if things look off and let
+                    # git-annex-init complain if it can't actually handle it.
+                    lgr.warning(
+                        "Expected an int for datalad.repo.version, got %s",
+                        version)
```

Comment:
I think here we might want to do something about `version` instead of just going forward here.  this code path is not tested so ramifications aren't immediately clear

Reply:
looking at the logic: so we will end up with some `version` which we alert about here but then do not announce about auto upgrades. I think that is ok, and we can proceed.